### ISSUE TITLE
Building charm++ with omp

### ISSUE THREAD

**Issue by msoroush** [UIUC-PPL/charm#2810]
I am trying to compile charm++ version 6.10.1 with omp, intel compiler 2019, and for multicore arch `./build charm++ multicore-linux-x86_64 omp iccstatic -j8 --with-production`, but I get the following error:
```
../bin/charmc  -optimize -production   -o ../lib/libck.a trace-common.o tracec.o tracef.o init.o register.o qd.o ck.o msgalloc.o ckfutures.o ckIgetControl.o debug-message.o debug-charm.o ckcallback.o cklocation.o ckmulticast.o ckarrayoptions.o ckarray.o ckreduction.o ckrdma.o waitqd.o LBDatabase.o LBManager.o MetaBalancer.o weakTest.o treeTest.o forestTest.o readmodel.o lbdbf.o ckobjQ.o ckcheckpoint.o ckmemcheckpoint.o ckevacuation.o ckobjid.o LBComm.o LBObj.o LBMachineUtil.o CentralPredictor.o BaseLB.o CentralLB.o HybridBaseLB.o DistBaseLB.o ckgraph.o LButil.o RefinerTemp.o Refiner.o manager.o ckset.o ckheap.o LBSimulation.o modifyScheduler.o charmProjections.o ckbitvector.o pathHistory.o controlPoints.o arrayRedistributor.o cp_effects.o trace-controlPoints.o mpi-interoperate.o ckregex.o sdag.o 
ar: creating ../lib/libck.a
../bin/charmc  -optimize -production   -I.   -c -o convcore.o convcore.C
convcore.C(2211): error: identifier "CsdTaskEnqueue" is undefined
    CsdTaskEnqueue((void*)token);
    ^
compilation aborted for convcore.C (code 2)
Fatal Error by charmc in directory /home/msoroush/Project/code/namd/charm/multicore-linux-x86_64-omp-icc/tmp
   Command icpc -fpic -DCMK_GFORTRAN -D_REENTRANT -I../bin/../include -D__CHARMC__=1 -I. -O2 -U_FORTIFY_SOURCE -fno-stack-protector -fopenmp -c convcore.C -o convcore.o returned error code 2
charmc exiting...
gmake: *** [<builtin>: convcore.o] Error 1
-------------------------------------------------
Charm++ NOT BUILT. Either cd into multicore-linux-x86_64-omp-icc/tmp and try
to resolve the problems yourself, visit
http://charm.cs.illinois.edu/
for more information. Otherwise, email the developers at charm@cs.illinois.edu
```

I think the reason for this error is that the flag `--enable-task-queue` would not be set in **build** when I have only omp option: https://github.com/UIUC-PPL/charm/blob/master/build#L612-L635

**Comment by matthiasdiener** [comment]
Pinging @sbak5 - does omp work without smp?

**Comment by rbuch** [comment]
I'm speculating a bit, but I think we can probably special case this and add the `--enable-task-queue` with the `omp` option for multicore since it's basically a single node SMP build without a comm thread.

**Comment by matthiasdiener** [comment]
@msoroush: I implemented a potential fix for this issue (#2817). Please let me know if this works for you.

**Comment by msoroush** [comment]
@matthiasdiener thanks, I used the same fix in order to compile the charm++ with OpenMP.
